newrelic/data_source_newrelic_synthetics_monitor.go

```diff
@@ -0,0 +1,83 @@
+package newrelic
+
+import (
+	"fmt"
+	"log"
+
+	synthetics "github.com/dollarshaveclub/new-relic-synthetics-go"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceNewRelicSyntheticsMonitor() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceNewRelicSyntheticsMonitorRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"api_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("NEWRELIC_API_KEY", nil),
+				Sensitive:   true,
+			},
+			"max_check": {
```

Comment:
Is there a reason to limit how many it queries? Why not just query them all? If the sorting isn't deterministic this could be problematic as it would result in erroneous diffs.

Reply:
There was a reason for this but I dont recall what it was at this point, I'll look into this again